File: C Language/Q17.c
#include <stdio.h>

int main() {
    double number = 15.58971;
    int integralPart;
    double fractionalPart;

    // Cast to int to get the whole number
    integralPart = (int)number;
    
    // Subtract the whole number from the original to get the fraction
    fractionalPart = number - integralPart;

    printf("Original Number: %f\n", number);
    printf("Integral Part: %d\n", integralPart);
    printf("Fractional Part: %f\n", fractionalPart);

    return 0;
}

File: Python Language/Q17.py
# Program to separate integral and fractional parts

# The number given
number = 15.58971

# Getting the integral part using int() casting
integral_part = int(number)

# Getting the fractional part by subtracting the integer part
fractional_part = number - integral_part

print(f"Original Number: {number}")
print(f"Integral part: {integral_part}")
print(f"Fractional part: {fractional_part}")